.github/workflows/yocto-build-deploy.yml

```
@@ -0,0 +1,1126 @@
+      #     if ! docker pull "${image_tag}"; then
+      #       image_tag="${HELPER_IMAGE_REPO}:${YOCTO_SCRIPTS_REF}-${HELPER_IMAGE_VARIANT}"
+      #       docker pull "${image_tag}"
+      #     fi
+
+      #     image_id="$(docker images --format "{{.ID}}" "${image_tag}")"
+      #     echo "id=${image_id}" >> $GITHUB_OUTPUT
+
+      # - name: Deploy AMI
+      #   if: inputs.deploy-ami == true
+      #   env:
+      #     AWS_DEFAULT_REGION: "${{ env.S3_REGION }}"
+      #     AWS_SESSION_TOKEN: "" # only required if MFA is enabled
+      #     AWS_SUBNET_ID: ${{ vars.AWS_SUBNET || 'subnet-02d18a08ea4058574' }}
+      #     AWS_SECURITY_GROUP_ID: ${{ vars.AWS_SECURITY_GROUP || 'sg-057937f4d89d9d51c' }}
+      #     BALENACLI_TOKEN: ${{ secrets.BALENA_API_KEY }}
+      #     HOSTOS_VERSION: "${{ steps.balena-lib.outputs.os_version }}"
+      #     AMI_NAME: "${{ steps.ami-name.outputs.string }}"
+      #     AMI_ARCHITECTURE: "${{ steps.ami-arch.outputs.string }}"
+      #     AMI_SECUREBOOT: "${{ inputs.sign-image }}"
+      #     BALENA_PRELOAD_APP: "balena_os/cloud-config-${{ steps.ami-arch.outputs.string }}"
+      #     BALENA_PRELOAD_COMMIT: current
+      #     IMAGE: ${{ runner.temp }}/deploy/image/balena.img
+      #   run: |
+      #     docker run --rm -t \
+      #       --privileged \
+      #       --network host \
+      #       -v "${WORKSPACE}:${WORKSPACE}" \
+      #       -v /var/run/docker.sock:/var/run/docker.sock \
+      #       -e VERBOSE \
+      #       -e AWS_ACCESS_KEY_ID \
+      #       -e AWS_SECRET_ACCESS_KEY \
+      #       -e AWS_DEFAULT_REGION \
+      #       -e AWS_SESSION_TOKEN \
+      #       -e AMI_NAME \
+      #       -e AMI_ARCHITECTURE \
+      #       -e AMI_SECUREBOOT \
+      #       -e S3_BUCKET \
+      #       -e BALENA_PRELOAD_APP \
+      #       -e BALENARC_BALENA_URL \
+      #       -e BALENACLI_TOKEN \
+      #       -e BALENA_PRELOAD_COMMIT \
+      #       -e IMAGE \
+      #       -e MACHINE \
+      #       -e HOSTOS_VERSION \
+      #       -e AWS_SUBNET_ID \
+      #       -e AWS_SECURITY_GROUP_ID \
+      #       -w "${WORKSPACE}" \
+      #       "${{ steps.ami-helper-image.outputs.id }}" /balena-generate-ami.sh
+
+      # Creates a test matrix to test a number of suites
+      # We do this to run each suite in parallel if there are available workers, and to be able to retry individual test suites
+      # Example output: + json='{"TEST_SUITE":["os","hup","cloud"],"DEVICE_TYPE":["generic-amd64"],"ENVIRONEMENT":["balenaos-balenamachine"],"WORKER_TYPE":["qemu"]}'
+      - name: Create Leviathan test matrix
+        id: leviathan_test_matrix
+        if: inputs.run-tests == true
+        env:
+          MATRIX: >
+            {
+              "TEST_SUITE": ${{ inputs.test-suites }},
+              "DEVICE_TYPE": ${{ format('["{0}"]', steps.balena-lib.outputs.device_slug) }},
+              "ENVIRONEMENT": ${{ format('["{0}"]', inputs.test-environment) }},
```

This one is also puzzling me. I believe this is a typo, but:

1. The example output above also says `ENVIRONEMENT` like here (possibly just copy and paste).
2. I couldn't spot where this is used to double check it.

```suggestion
              "ENVIRONMENT": ${{ format('["{0}"]', inputs.test-environment) }},
```